### ISSUE TITLE
Plugin throws exception with Jetpack Compose apps

### ISSUE THREAD

**Issue by sjmueller** [bugsnag/bugsnag-android-gradle-plugin#354]
When used within a Jetpack Compose project, like so:
```
plugins {
    id "com.android.application"
    id "kotlin-android"
    id "com.bugsnag.android.gradle" version "5.6.0"
}
```

Gradle sync fails with:

```
Unable to find method ''void com.android.build.api.dsl.CommonExtension.onVariants(kotlin.jvm.functions.Function1)''
'void com.android.build.api.dsl.CommonExtension.onVariants(kotlin.jvm.functions.Function1)'
```

The current requirements for Jetpack Compose are using Android Studio Arctic Fox Canary 3 with kotlin `1.4.20` and a dependency on `classpath "com.android.tools.build:gradle:7.0.0-alpha03"` (which seems to end up using `6.8-rc-1` in `.gradle` directory).

This could be a similar problem to #319, but in the case of Jetpack Compose there isn't an option to use the stable path. Given these hard requirements, can you share precise instructions on a recommended alternative (manual?) approach to using bugsnag in a Jetpack Compose app?

**Comment by yousif-bugsnag** [comment]
Hi @sjmueller 

Jetpack Compose isn't something we officially support, but I've raised it with our product team as something we may wish to look at. Our gradle plugin also does not yet support AGP 7.x. This is on our roadmap to provide support for in a future version.

**Comment by sjmueller** [comment]
Thanks for the feedback @yousif-bugsnag, can you share with me specific ways that bugsnag can be used properly without the gradle plugin? Is it an absolute requirement to use it or are there manual workarounds..

**Comment by mattdyoung** [comment]
Hi @sjmueller 

It's perfectly possible to use the Bugsnag Android integration without the Bugsnag gradle plugin.

You'd just need to add the following dependency to `<project_dir>/app/build.gradle`:
```
dependencies {
    // ...
    implementation "com.bugsnag:bugsnag-android:5.+"
}
```

And then follow the steps under _Basic Configuration_ here:
https://docs.bugsnag.com/platforms/android/#basic-configuration

The Bugsnag gradle plugin automates the upload of mapping files and reporting of build version information, so if you're not using the plugin we'd recommend performing these actions manually using the relevant APIs:
https://docs.bugsnag.com/api/android-mapping-upload/
https://docs.bugsnag.com/api/ndk-symbol-mapping-upload/
https://docs.bugsnag.com/api/build/

**Comment by johnaziz57** [comment]
Currently in our project we are not use `Jetpack Compose` but we are getting the same error
Gradle 6.7.1 and AGP 4.2.1
This is currently hindering out efforts to upgrade to the latest Gradle plugin

Can provide more information if needed

**Comment by yousif-bugsnag** [comment]
Hi @johnaziz57, could you please write in to support@bugsnag.com with the full details of the error you're seeing and the version of `bugsnag-android-gradle-plugin` you're using?

**Comment by johnkiely1** [comment]
Hi @sjmueller 

We've now released [v7.0.0-beta01](v7.0.0-beta01) of the plugin which offers AGP7 Support.
There is an upgrade guide available [here](https://github.com/bugsnag/bugsnag-android-gradle-plugin/blob/v7.0.0-beta01/UPGRADING.md) if you wanted to try see how you get on with that as this was a previous blocker for Jetpack Compose. 

Do bear in mind this is currently a beta as AGP7 is still in beta.

**Comment by mattdyoung** [comment]
Just an update to say that AGP7 is now fully supported in [v7.x](https://github.com/bugsnag/bugsnag-android-gradle-plugin/releases/tag/v7.1.0) versions of bugsnag-android-gradle-plugin.
